cdb2api/cdb2api.c

```diff
@@ -4763,7 +4763,7 @@ int cdb2_open(cdb2_hndl_tp **handle, const char *dbname, const char *type,
         char *p;
         p = strchr(type, ':');
         if (p) {
-            strcpy(hndl->hosts[0], type);
+            strncpy(hndl->hosts[0], type, sizeof(hndl->hosts[0]));
```

Comment:
@adizaimi, Indeed, however we still need to explicitly nul-terminate: something like `strlcpy` will get us both in one call.

We can probably perform better validation elsewhere, but I'm still familiarizing myself with the code. I'll patch this with explicit nul termination.